### PR TITLE
Processing pipeline

### PR DESCRIPTION
**Pull request by darrylmasson**
Ideally this is a replacement for Storm. It's a bit on the bare-bones side now, but have to start somewhere. Still things marked `#TODO`, most of which should be addressed before this gets merged. Some documentation so far [on the wiki](https://wiki.uni-freiburg.de/app/doku.php?id=start:ultimate:slow_control:pipeline)

### CODE REVIEW

**Comment by darrylmasson** [comment]
Also still needs doing:
- Handle things relating to time (ie, polling frequency, lengths of buffers for IntegralNode and DifferentialNode)
- Write PipelineMonitor wrapper for Doberman to work with

File: Doberman/AlarmMonitor.py [hunk 1]
```diff
@@ -230,11 +230,14 @@ def check_shifters(self):
         now = dtnow()
         shift = self.db.read_from_db('settings', 'shifts',
                 {'start': {'$lte': now}, 'end': {'$gte': now}}, onlyone=True)
+        if shift == None:
+            self.current_shifters = []
+            return
         new_shifters = shift['shifters']
         new_shifters.sort()
         shift_end = shift['end']
         if new_shifters != self.current_shifters and len(''.join(new_shifters)) != 0:
```
Comment (adambrown1) [inline Doberman/AlarmMonitor.py:239]: There should be an alarm to the old shifters if there are no new shifters

Comment (adambrown1) [reply Doberman/AlarmMonitor.py:239]: Oops, maybe this is not the right place for this comment :D

Comment (darrylmasson) [reply Doberman/AlarmMonitor.py:239]: The comment is valid but indeed not really related to this PR. Probably better to open an issue.

**Comment by jarongrigat** [comment]
Merged to experimental rather than to master

**Pull request closed without merging**